ai/docs/GPT Online Model Training Documentation .md

```diff
@@ -0,0 +1,312 @@
+```
+
+For OpenAI official tutorial, please [refer to this](https://openai.com/blog/gpt-3-5-turbo-fine-tuning-and-api-updates), or [more specific one](https://platform.openai.com/docs/guides/fine-tuning).
+
+# Steps
+
+1. Prepare formatted data
+The data are prepared to **jsonl**format, where each line is an effective json object, representing a training sample. An example of official formatted training sample for gpt-3.5-turbo fine-tuning is like this.
+
+```plain
+{
+  "messages": [
+    { "role": "system", "content": "You are an assistant that occasionally misspells words" },
+    { "role": "user", "content": "Tell me a story." },
+    { "role": "assistant", "content": "One day a student went to schoool." }
+  ]
+}
+```
+The "content" of the "system" is the systematic prompt describes what general task you want the model to do, or which format you want the model to output. The "content" of the "user" is the specific task you want the model to do. The "content" of the "assistant" is the expected model outputs. 
+You need to adjust your data format into the official one. Suppose you have a raw data format like this, named "input.json":
+
+```plain
+ {
+  "id": "identity_1",
+  "conversations": [
+   {
+    "from": "human",
+    "value": "title: Egress should not be applied to traffic destined for ServiceCIDRs. body: **Describe the bug**\r\n\r\nThe issue was reported by @sroot1986, @claknee, @robbo10:\r\n\r\nWhen AntreaProxy is asked to skip some Services by configuring `skipServices` or is not running at all, the traffic destined for Service ClusterIP is supposed to reach host network and be proxied by kube-proxy, or be processed by the host directly (e.g. [NodeLocal DNSCache](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/)). However, if Egress is applied to the Pod, the traffic would be forwarded to Egress Node and would be proxied remotely, as opposed to locally, which could incur  performance issue and unexpected behaviors.\r\n\r\n\r\n**To Reproduce**\r\n\r\n\r\n1. Install [NodeLocal DNSCache](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/)\r\n2. Configure AntreaProxy to skip DNS service, following [the doc](https://github.com/antrea-io/antrea/blob/main/docs/antrea-proxy.md#when-you-are-using-nodelocal-dnscache)\r\n3. Apply Egress to a Pod, which is scheduled to a Node different from the Egress Node.\r\n4. Trigger DNS query from the Pod, the Node local DNS server won't receive the request and the resolution may fail\r\n\r\n**Expected**\r\n\r\n\r\nPod-to-Service ClusterIP should not be forwarded to Egress Node and should be handled locally regardless of how AntreaProxy is configured.\r\n\r\n**Versions:**\r\n\r\n - Antrea version (Docker image tag). v1.11.3, v1.12.2, v1.13.1\r\n\r\n"
+   },
+   {
+    "from": "gpt",
+    "value": "['transit']"
+   }
+  ]
+ },
+```
+You can run the python script named "data_process.py" to convert it into the right form named "output.jsonl".
+```plain
```

Comment:
Specify whether the mentioned Python packages (openai, pathlib) need to be installed separately or if they are included in the OpenAI Python environment.

Reply:
updated